fp16_speedup

rydenisbak opened the pull request:
PR from this issue
https://github.com/open-mmlab/mmdetection/issues/2131

hellock commented on mmdet/core/fp16/hooks.py:
```diff
@@ -97,8 +97,12 @@ def wrap_fp16_model(model):
+        path_forward_condition = isinstance(module, nn.GroupNorm)
+        major, minor = [int(v) for v in torch.__version__.split('.')][:2]
+        path_forward_condition |= major < 1 or major == 1 and minor < 3
+        if path_forward_condition:
```
The condition can be rewritten in just one line:
```python
if torch.__version__ < (1, 3) or isinstance(module, nn.GroupNorm):
    module.forward = patch_forward_method(module.forward, torch.half, torch.float)
```

rydenisbak replied:
![image](https://user-images.githubusercontent.com/16317197/75418899-a7242200-5945-11ea-81dc-3a83c0e9961e.png)
but I can write
![image](https://user-images.githubusercontent.com/16317197/75418952-c4f18700-5945-11ea-857b-b4abe590ff56.png)
It looks more pretty, thanks

hellock commented:
Thanks! Could you fix the linting error?

rydenisbak commented:
@hellock I found some bug in the code.
My torch version is '1.3.1+cu100' and it cause problem in line
https://github.com/rydenisbak/mmdetection/blob/fp16_speedup/mmdet/core/fp16/hooks.py#L100
```python
torch_version = [int(v) for v in torch.__version__.split('.')]
```
We need to change it, for example like this
```python
torch_version = [int(v) for v in torch.__version__.split('.') if v.isdigit()]
```
or like this
```python
torch_version = [int(v) if v.isdigit() else int(v.split('+')[0]) for v in torch.__version__.split('.') if v.isdigit()]
```
or like something else

hellock commented:
Oh the pytorch versions installed by pip and conda are different. Maybe simply
```python
torch_version = [int(v) for v in torch.__version__.split('.')[:2]]
```

rydenisbak commented:
ok, good
